lib/improver/tests/blending/weighted_blend/test_WeightedBlendAcrossWholeDimension.py

```diff
@@ -345,59 +508,235 @@ def test_percentiles_different_coordinate_orders(self):
+        expected = np.full((2, 2), 0.6)
+
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
 
     @ManageWarnings(
         ignored_messages=["Collapsing a non-contiguous coordinate."])
-    def test_weighted_max_non_equal_weights_list(self):
-        """Test it works for weighted_max with weights [0.2, 0.8]
-           given as a list."""
-        coord = "time"
+    def test_with_spatially_varying_weights(self):
+        """Test function when a data cube and a multi dimensional weights cube
+        are provided. This tests spatially varying weights, where each x-y
+        position is weighted differently in each slice along the blending
+        coordinate."""
+
+        coord = "forecast_reference_time"
+        plugin = WeightedBlendAcrossWholeDimension(coord, 'weighted_mean')
+        result = plugin.weighted_maximum(self.cube, self.weights3d)
+        expected = np.array([[2.4, 1.2], [1.8, 0.8]])
+
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    @ManageWarnings(
+        ignored_messages=["Collapsing a non-contiguous coordinate."])
+    def test_without_weights(self):
+        """Test function when a data cube is provided, but no weights cube
+        which should result in equal weightings."""
+
+        coord = "forecast_reference_time"
         plugin = WeightedBlendAcrossWholeDimension(coord, 'weighted_maximum')
-        weights = [0.2, 0.8]
-        result = plugin.process(self.cube, weights)
-        expected_result_array = np.ones((2, 2))*1.6
-        self.assertArrayAlmostEqual(result.data, expected_result_array)
+        result = plugin.weighted_maximum(self.cube, None)
+        expected = np.full((2, 2), 1.)
+
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
 
     @ManageWarnings(
         ignored_messages=["Collapsing a non-contiguous coordinate."])
-    def tests_threshold_splicing_works_weighted_max(self):
-        """Test weighted_max works with a threshold dimension."""
+    def tests_threshold_cube_with_weights_blending_threshold(self):
+        """Test weighted_mean works collapsing a cube with a threshold
+        dimension when the blending is over the threshold coordinate. In this
+        case there is no slicing over the threshold coordinate, so we can test
+        the functionality at this level."""
+        coord = "threshold"
+        plugin = WeightedBlendAcrossWholeDimension(coord, 'weighted_mean')
+        result = plugin.weighted_maximum(self.cube_threshold,
+                                         self.weights_threshold)
+
+        expected_data = np.array([0.16, 0.32, 0.48])
+        expected_array = np.broadcast_to(expected_data, (2, 2, 3)).T
+        self.assertArrayAlmostEqual(result.data, expected_array)
+
+
+class Test_process(Test_weighted_blend):
```

Comment:
It would be good if there was an additional test that included using the `cycletime` and `coords_for_bounds_removal` options, if we're keeping this option, to check that these options work as intended.

Reply:
I have added the ```test_specific_cycletime``` test to cover using cycletime. The coords_for_bounds_removal bits have been removed.